src/lib/Scenes/Artwork/Artwork.tsx

```diff
@@ -1,47 +1,126 @@
-import { Flex, Join, Spacer, Theme } from "@artsy/palette"
+import { Box, Theme } from "@artsy/palette"
 import { Artwork_artwork } from "__generated__/Artwork_artwork.graphql"
 import { ArtworkQuery } from "__generated__/ArtworkQuery.graphql"
 import Separator from "lib/Components/Separator"
 import { defaultEnvironment } from "lib/relay/createEnvironment"
 import renderWithLoadProgress from "lib/utils/renderWithLoadProgress"
 import React from "react"
-import { Dimensions, ScrollView } from "react-native"
+import { Dimensions, FlatList } from "react-native"
 import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"
 import { AboutArtistFragmentContainer as AboutArtist } from "./Components/AboutArtist"
-import { AboutWorkFragmentContainer as AboutWork } from "./Components/AboutWork"
-import { ArtworkActionsFragmentContainer as ArtworkActions } from "./Components/ArtworkActions"
-import { ArtworkAvailabilityFragmentContainer as ArtworkAvailability } from "./Components/ArtworkAvailability"
+import { AboutWork } from "./Components/AboutWork"
+import { ArtworkAvailability } from "./Components/ArtworkAvailability"
 import { ArtworkDetailsFragmentContainer as ArtworkDetails } from "./Components/ArtworkDetails"
+import { ArtworkHeader } from "./Components/ArtworkHeader"
 import { ArtworkTombstoneFragmentContainer as ArtworkTombstone } from "./Components/ArtworkTombstone"
-import { ImageCarouselFragmentContainer as ImageCarousel } from "./Components/ImageCarousel/ImageCarousel"
 import { OtherWorksFragmentContainer as OtherWorks } from "./Components/OtherWorks"
-import { SellerInfoFragmentContainer as SellerInfo } from "./Components/SellerInfo"
+import { SellerInfo } from "./Components/SellerInfo"
 
 interface Props {
   artwork: Artwork_artwork
 }
 
+interface State {
+  sections: Array<{
+    type: "tombstone" | "availability" | "sellerInfo" | "aboutWork" | "details" | "aboutArtist" | "otherWorks"
+    data: any
+  }>
+}
+
 export class Artwork extends React.Component<Props> {
-  render() {
+  state: State = {
+    sections: [],
+  }
+
+  componentDidMount() {
```

Comment:
@l2succes can you elaborate on why we need to append the sections in `componentDidMount` vs. in the render function?

Reply:
+1 especially because the `setState` call will trigger a new render pass immediately after the one that led to `componentDidMount`.

Reply:
no need to use `setState` here, it definitely makes more sense to have it in the render function.